libass/ass_fontselect.c

```diff
@@ -1000,8 +1000,9 @@ ass_embedded_fonts_add_provider(ASS_Library *lib, ASS_FontSelector *selector,
         load_fonts_from_dir(lib, lib->fonts_dir);
     }
 
-    for (size_t i = 0; i < lib->num_fontdata; ++i)
+    for (size_t i = lib->num_fontsloaded; i < lib->num_fontdata; ++i)
```

Comment:
Do I understand correctly that this will always be zero here? The only other place that increments `num_fontsloaded` is in `ass_update_embedded_fonts`, which assumes the embedded font provider already exists.

Reply:
My thought was that if `ass_set_fonts` is called a second time *(is that allowed API use?)*, already loaded fonts won't be added a second time. However, looking at it again, the embedded font provider gets recreated so it needs to start from zero regardless.

`ass_update_embedded_fonts` should also check that the provider does exists, in case creation failed.

Re-pushed with both of the above changes.